internal/pkg/agent/install/install.go

```diff
@@ -53,15 +58,36 @@ func Install(cfgFile string) error {
 
 	// place shell wrapper, if present on platform
 	if paths.ShellWrapperPath != "" {
-		err = os.MkdirAll(filepath.Dir(paths.ShellWrapperPath), 0755)
-		if err == nil {
-			err = ioutil.WriteFile(paths.ShellWrapperPath, []byte(paths.ShellWrapper), 0755) // nolint:gosec,G306 // this is fine.
-		}
-		if err != nil {
-			return errors.New(
-				err,
-				fmt.Sprintf("failed to write shell wrapper (%s)", paths.ShellWrapperPath),
-				errors.M("destination", paths.ShellWrapperPath))
+		// Install symlink for darwin instead
+		if runtime.GOOS == darwin {
+			// Check if previous shell wrapper or symlink exists and remove it so it can be overwritten
+			if _, err := os.Lstat(paths.ShellWrapperPath); err == nil {
+				if err := os.Remove(paths.ShellWrapperPath); err != nil {
+					return errors.New(
+						err,
+						fmt.Sprintf("failed to remove (%s)", paths.ShellWrapperPath),
+						errors.M("destination", paths.ShellWrapperPath))
+				}
+			}
+			err = os.Symlink("/Library/Elastic/Agent/elastic-agent", paths.ShellWrapperPath)
```

Comment:
replace with:
`filepath.Join(paths.InstallPath, paths.BinaryName, paths.ShellWrapper)`

Reply:
will leave as is for now, even though the change is trivial, because:
1. previously this path was hardcode in the shell wrapper script as is
```
	// ShellWrapper is the wrapper that is installed.
	ShellWrapper = `#!/bin/sh
exec /Library/Elastic/Agent/elastic-agent $@
`
```
2. this is a platform specific for Darwin so path separators and the path are fixed

can change (and would have to retest) when merging to the main branch, not for the urgent fix